api/spec/v1.yaml

```diff
@@ -771,6 +773,12 @@ paths:
             best-effort basis. For example, it inspects ERC20 methods inside `evm.Call` txs.
             However, you must provide the oasis-style derived address here, not the Eth address.
             See `AddressPreimage` for more info on oasis-style vs Eth addresses.
+        - in: query
```

Comment:
Second option gets into the auto detecting search territory?

Reply:
I prefer the second option as well; we were planning on adding support for eth hashes for a while anyway :)

> detect which hash the user intended (with heuristics

AFAIK the hashes are indistinguishable - they're both 32-byte hashes so I'll go ahead with the `WHERE (hash=$1 OR eth_hash=$1)` like you suggested

> BTW, it might make sense to tackle https://app.clickup.com/t/861maqcru

Yup will do

Reply:
> Second option gets into the auto detecting search territory?

It slightly does. I'm okay with this since the logic isn't prescriptive; we do not try to infer the 'type' of tx desired and limit the returned tx types. Instead we simply return _all_ transactions with any matching hash. I also tried to make the openapi description as descriptive as possible.

Though this does raise the question; do we want to support searching by _only_ tx_hash or tx_eth_hash? This would be nice for the block explorer since users will click into any transaction and see the transaction they expect. I suppose we can rely on the fact that collisions wouldn't occur... in theory we'd need ~2^128 hashes before realistically seeing a collision